app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -535,4 +565,71 @@ public CustomProgressDialog generateProgressDialog(int taskId) {
     public boolean isBackEnabled() {
         return false;
     }
+    
+    private void refreshView() {
+        // Refresh the breadcrumb bar in case the seated app has changed
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
+            refreshBreadcrumbBar();
+        }
+
+        // Decide whether or not to show the app selection spinner based upon # of usable apps
+        ArrayList<ApplicationRecord> readyApps = CommCareApplication._().getUsableAppRecords();
+        if (readyApps.size() == 1) {
+            spinner.setVisibility(View.GONE);
+            welcomeMessage.setText(Localization.get("login.welcome.single"));
+            // Set this app as the last selected app, for use in choosing what about to initialize
+            // on first startup
+            ApplicationRecord r = readyApps.get(0);
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+            prefs.edit().putString(KEY_LAST_APP, r.getUniqueId()).commit();
+        }
+        else {
+            welcomeMessage.setText(Localization.get("login.welcome.multiple"));
+            ArrayList<String> appNames = new ArrayList<>();
+            ArrayList<String> appIds = new ArrayList<>();
+            for (ApplicationRecord r : readyApps) {
+                String name = r.getDisplayName();
+                appNames.add(name);
+                appIds.add(r.getUniqueId());
+                namesToRecords.put(name, r);
+            }
+            ArrayAdapter<String> adapter = new ArrayAdapter<>(this, R.layout.spinner_text_view, appNames);
+            adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+            spinner.setAdapter(adapter);
+            spinner.setOnItemSelectedListener(this);
+            // Set the spinner's selection to match whatever the currently seated app is
+            String currAppId = CommCareApplication._().getCurrentApp().getUniqueId();
+            int position = appIds.indexOf(currAppId);
+            spinner.setSelection(position);
+            spinner.setVisibility(View.VISIBLE);
+        }
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        // Retrieve the app record corresponding to the app selected
+        String selected = (String) parent.getItemAtPosition(position);

```

Comment:
Kind of sketchy to do this by name, btw. Doing it by index is likely OK (IE: Make an ArrayList of AppRecords in the same order as the items)